app/move/controllers/unassign.test.js

```diff
@@ -0,0 +1,277 @@
+const proxyquire = require('proxyquire')
+
+const FormWizardController = require('../../../common/controllers/form-wizard')
+
+const allocationGetByIdStub = sinon.stub()
+allocationGetByIdStub.resolves({ id: '__allocation__' })
+const moveUnassignStub = sinon.stub()
+moveUnassignStub.resolves({})
+
+const UnassignController = proxyquire('./unassign', {
```

Comment:
You could drop the need for proxyquire altogether here and stub the services using:

```javascript
const allocationService = require('../../../common/services/allocation')

...

beforeEach(function() {
  sinon.stub(allocationService, getById)
})
```

If find this method much cleaner to use where possible. Then you can also drop the need for having to do things like `. resetHistory()` as it's already being done globally with sinon.

Reply:
It literally wasn't working when I did it the way you're suggesting previously. But now it works.

I've made the change but I do disagree. It's only because of mocha's lack of parallelisation that this works and also it works for some things and not for others.  (*cough* https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/issues/495)

Reply:
Yeah I'd actually be keen to explore rewire. I'm not really a fan of proxyquire and that's why I try and avoid using it where we can and use `sinon.stub()`. It makes it easier to change the behaviour in nested tests of a stub, otherwise you have to proxyquire within a `beforeEach` if sending the stub in via proxyquire. And that gets...well, messy very quickly.